src/Sarif/Autogenerated/ReportingDescriptor.cs

```diff
@@ -56,13 +56,13 @@ public virtual SarifNodeKind SarifNodeKind
         /// A unique identifier for the reporting descriptor in the form of a GUID.
         /// </summary>
         [DataMember(Name = "guid", IsRequired = false, EmitDefaultValue = false, Order = 10)]
-        public virtual string Guid { get; set; }
+        public virtual Guid? Guid { get; set; }
 
         /// <summary>
         /// An array of unique identifies in the form of a GUID by which this report was known in some previous version of the analysis tool.
         /// </summary>
         [DataMember(Name = "deprecatedGuids", IsRequired = false, EmitDefaultValue = false, Order = 11)]
-        public virtual IList<string> DeprecatedGuids { get; set; }
+        public virtual IList<Guid> DeprecatedGuids { get; set; }
```

Comment:
>DeprecatedGuids [](http://example.com/codeflow?start=35&length=15)

Just curious, do we have an explicit test that populates, serializes this list of Guids? I think we do. It just occurs to me this is a pretty specialized code path/utilization, these list of struct Guids...

Reply:
yes we did have, and I made those test data from string to Guid in the list.